android/src/main/java/com/oney/WebRTCModule/DailyWebRTCDevicesManager.java

```diff
@@ -0,0 +1,201 @@
+            enumerateDevicesArray.pushMap(params);
+        }
+    }
+
+    private void fillAudioDevices(WritableArray enumerateDevicesArray){
+        AudioDeviceInfo[] audioOutputDevices = this.audioManager.getDevices(AudioManager.GET_DEVICES_OUTPUTS);
+
+        boolean isWiredHeadsetPlugged = Arrays.stream(audioOutputDevices).anyMatch(
+                device -> device.getType() == AudioDeviceInfo.TYPE_WIRED_HEADSET ||
+                        device.getType() == AudioDeviceInfo.TYPE_WIRED_HEADPHONES
+        );
+
+        WritableMap params = isWiredHeadsetPlugged ?
+                this.createWritableMap(AudioDeviceType.WIRED_OR_EARPIECE.toString(), "Wired headset", DeviceKind.AUDIO.getKind()) :
+                this.createWritableMap(AudioDeviceType.WIRED_OR_EARPIECE.toString(), "Phone Earpiece", DeviceKind.AUDIO.getKind());
+        enumerateDevicesArray.pushMap(params);
+
+        //speaker
+        params = this.createWritableMap(AudioDeviceType.SPEAKER.toString(), "Speaker", DeviceKind.AUDIO.getKind());
+        enumerateDevicesArray.pushMap(params);
+
+        boolean isBluetoothHeadsetPlugged = Arrays.stream(audioOutputDevices).anyMatch(device -> device.getType() == AudioDeviceInfo.TYPE_BLUETOOTH_SCO);
+        if(isBluetoothHeadsetPlugged){
+            params = this.createWritableMap(AudioDeviceType.BLUETOOTH.toString(), "Bluetooth", DeviceKind.AUDIO.getKind());
+            enumerateDevicesArray.pushMap(params);
+        }
+    }
+
+    private WritableMap createWritableMap(String deviceId, String label, String kind){
+        WritableMap audioMap = Arguments.createMap();
+        audioMap.putString("deviceId", deviceId);
+        audioMap.putString("groupId", "");
+        audioMap.putString("label", label);
+        audioMap.putString("kind", kind);
+        return audioMap;
+    }
+
+    /** Changes selection of the currently active audio device. */
+    public void setAudioDevice(String deviceId) {
+        Log.d(TAG, "setAudioDevice(audioDeviceType=" + deviceId + ")");
+        AudioDeviceType audioRoute = AudioDeviceType.valueOf(deviceId);
+        switch (audioRoute) {
+            case SPEAKER:
+                toggleBluetooth(false);
+                audioManager.setSpeakerphoneOn(true);
+                break;
+            //If we have a wired headset plugged, It is not possible we send the audio to the earpiece
+            case WIRED_OR_EARPIECE:
+                toggleBluetooth(false);
+                audioManager.setSpeakerphoneOn(false);
+                break;
+            case BLUETOOTH:
+                audioManager.setSpeakerphoneOn(false);
+                toggleBluetooth(true);
+                break;
+            default:
+                Log.e(TAG, "Invalid audio device selection");
+                break;
+        }
+    }
+
+    public String getAudioDevice() {
+        if(this.audioManager.isBluetoothScoOn() || this.audioManager.isBluetoothA2dpOn()){
```

Comment:
Super-nit: can you get your IDE to auto-format these new files? I've seen some inconsistent spacing around, for example, curly braces.

Reply:
I have not found a way to Android Studio automatically format on save. But I have formatted both the files (`Option` + `Command` + `L`)
Try to find out that later, If I can install some pluggin on Android Studio to apply that automatically.